examples/source_separation/conv_tasnet/README.md

```diff
@@ -0,0 +1,60 @@
+# Conv-TasNet
+
+This is a reference implementation of Conv-TasNet.
+
+> Luo, Yi, and Nima Mesgarani. "Conv-TasNet: Surpassing Ideal Time-Frequency Magnitude Masking for Speech Separation." IEEE/ACM Transactions on Audio, Speech, and Language Processing 27.8 (2019): 1256-1266. Crossref. Web.
+
+This implementation is based on [arXiv:1809.07454v3](https://arxiv.org/abs/1809.07454v3) and [the reference implementation](https://github.com/naplab/Conv-TasNet) provided by the authors.
+
+For the usage, please checkout the [source separation README](../README.md).
+
+## (Default) Training Configurations
+
+The default training/model configurations follows the best non-causal implementation from the paper. (causal configuration is not implemented.)
+
+ - Sample rate: 8000 Hz
+ - Batch size: total 32 over distributed training workers
+ - Epochs: 100
+ - Initial learning rate: 1e-3
+ - Gradient clipping: maximum L2 norm of 5.0
+ - Optimizer: Adam
+ - Learning rate scheduling: Halved after 3 epochs of no improvement in validation accuracy.
+ - Objective function: SI-SNRi (The paper uses SI-SNR as the objective function, which differs only by a constant value, so it should practically yield the same result.)
+ - Reported metrics: SI-SNRi, SDRi
+ - Sample audio length: 4 seconds (first 4 seconds)
+ - Encoder/Decoder feature dimension (N): 512
+ - Encoder/Decoder convolution kernel size (L): 16
+ - TCN bottleneck/output feature dimension (B): 128
+ - TCN hidden feature dimension (H): 512
+ - TCN skip connection feature dimension (Sc): 128
+ - TCN convolution kernel size (P): 3
+ - The number of TCN convolution block layers (X): 8
+ - The number of TCN convolution blocks (R): 3
+
+## Training
+
+The training takes about 5 mins per epoch with 8 V100 GPUs in a single node.
+
+## Evaluation
+
+The following is the evaluation result of training the model on WSJ0-2mix and WSJ0-3mix datasets.
+
+### wsj0-mix 2speakers
```

Comment:
Do you want to upload a model to make it easier to reproduce these numbers?

Reply:
Yeah, I think that will be great.